frontend/src/entities/jobs/algorithms.tsx

```diff
@@ -71,83 +73,114 @@ export type Hyperparameter = {
-    const number_context = z.number().min(minimum, {
-        message: `${positiveIntMessage(fieldName, required)} of at least ${minimum}`,
-    });
+/**
+ * Creates a validator for ensuring a number meets the required conditions
+ * 
+ * @param fieldName Name of the field being validated
+ * @param inputProps Properties that ensure the validity of the field
+ * @returns A validator for the field that ensures the desired requirements
+ */
+const numberValidator = (
+    fieldName: string,
+    inputProps?: rangeValidatorProps
+) => {
+    // Create properties by merging the defaults with the provided input properties
+    const props = {
+        min: Number.NEGATIVE_INFINITY, // Specifies the smallest number allowed (default is infinitely negative to allow for any number)
+        max: Number.POSITIVE_INFINITY, // Specifies the largest number allowed (default is ifinitely positive to allow for any number)
+        isFloat: false, // The number is assumed to be an Integer unless specified (default is that it is an Integer)
+        required: false, // Specify whether this field is required for submission and must not be empty (default is that the field is not required)
+        allowAuto: false, // Allows for the text option of "auto" instead of a number (default is that "auto" is not valid)
+        includeMin: true, // Whether the specified minimum is a valid number option (default is that it is valid)
+        includeMax: true, // Whether the specified maximum is a valid number option (default is that it is valid)
+        ...inputProps
+    }
 
-    let context: any = z.preprocess(Number, number_context);
-    if (!required) {
-        context = context.optional().or(z.literal(''));
+    // Create range message
+    let rangeMessage = '';
+    if (props.min !== Number.NEGATIVE_INFINITY && props.max !== Number.POSITIVE_INFINITY){
+        // Create range message
+        rangeMessage = ` value in the range ${props.includeMin?'[':'('}${props.min},${props.max}${props.includeMax?']':')'}`
+    } else if (props.max === Number.POSITIVE_INFINITY) {
+        // If there is a min, but not a max, create greater than message
+        rangeMessage = ` value greater than ${props.includeMin?'or equal to ':''}${props.min}`
+    } else if (props.min === Number.NEGATIVE_INFINITY) {
+        // If there is a max, but not a min, create less than message
+        rangeMessage = ` value less than ${props.includeMax?'or equal to ':''}${props.max}`
     }
-    return context;
-};
 
-const rangeValidator = (
-    fieldName: string,
-    min: number,
-    max: number,
-    isFloat = false,
-    required = true,
-    allowAuto = false
-) => {
-    const validationMessage = `${fieldName} must be ${
-        isFloat ? 'a float' : 'an integer'
-    } value in the range [${min},${max}]${allowAuto ? ' or be \'auto\'' : ''}`;
+    // Create validation message for errors
+    const validationMessage = `${fieldName}${props.required ? ' is required and' : '' } must be ${
```

Comment:
this variable looks extremely convoluted for it to be a "1 liner". Can this be split up into multiple lines to make it more readable?

Reply:
Yeah we can try. This was complex before, but I can try to make it simpler.